src/Tools/dotnet-dev-certs/src/Program.cs

```diff
@@ -361,25 +352,34 @@ private static int EnsureHttpsCertificate(CommandOption exportPath, CommandOptio
 
             if (trust?.HasValue() == true)
             {
-                if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                if (!manager.SupportsTrust)
```

Comment:
We want to make this check based on a static list of supported Distro/Version pairs. I don't want us to say we support `trust` if we don't support it at all levels (Firefox, Chrome/Debian, System); and as it is right now, this will always return true.

An alternative would be to make sure we print a warning when we detect a condition where we can't "guarantee" that we support trust, for example, if Edge/Chrome is not installed or we don't know how to trust the certificate for a given distro/version pair we can still support it even if partially and print a warning explaining this to the user.

Reply:
Its a priority for us that we are **VERY VERY** clear on what's supported here and what not. Otherwise the cost of supporting this skyrockets.

Reply:
> We want to make this check based on a static list of supported Distro/Version pairs.

Such conditions have been problematic in the past. For example, something works on a supported Fedora 33 version. The user upgrades to Fedora 34 and suddenly things stop working.

> An alternative would be to make sure we print a warning when we detect a condition where we can't "guarantee" that we support trust, for example, if Edge/Chrome is not installed or we don't know how to trust the certificate for a given distro/version pair we can still support it even if partially and print a warning explaining this to the user.

Yes, this is preferable imo.

It's valuable to make the browsers trust the cert, even if it is not know how to install for System.
Also the latter requires `root`, which the user may not have.
Trusting some is better than doing _all or nothing_.

Should this warning be generated using the `EventLog`?